consensus/istanbul/core/roundstate.go

```diff
@@ -177,6 +182,37 @@ func (s *roundStateImpl) Round() *big.Int {
 	return s.round
 }
 
+func (s *roundStateImpl) changeRound(nextRound *big.Int, validatorSet istanbul.ValidatorSet) {
+	s.state = StateAcceptRequest
+	s.round = nextRound
+	s.desiredRound = nextRound
```

Comment:
Are you sure it's correct to overwrite `desiredRound` here?

Reply:
`StartNewRound()` and `StartNewSequence` are replacements for `c.current := newRoundState()`. So basically in this impl i'm making sure i do the same things the constructor did.

Wether that is correct or not, not sure. But the impl should be equal.

There's a TODO from @trianglesphere about keeping the desired round on new rounds.